src/pages/index.js

```diff
@@ -5,12 +5,208 @@ import { PopupWithForm } from "../components/PopupWithForm.js";
 import { PopupWithImage } from "../components/PopupWithImage.js";
 import { Section } from "../components/Section.js";
 import { UserInfo } from "../components/UserInfo.js";
+import { Api } from "../components/Api.js";
+import { PopupDeleteConfirm } from "../components/PopupDeleteConfirm.js";
 import utils from "../Utils/Constants.js";
 import "../pages/index.css";
 
-/*----------------------------------------------------------*/
-/*               			FUNCTIONS		                */
-/*----------------------------------------------------------*/
+//USER INFO
+const userInfo = new UserInfo({
+	titleSelector: utils.htmlIds.profileTitle,
+	descriptionSelector: utils.htmlIds.profileDescription,
+	profilePictureSelector: ".profile__image",
+});
+
+const deleteCardModal = new PopupDeleteConfirm("#js-card-delete-modal");
+
+//need to make a change picture pop up html
+const editProfilePictureForm = new PopupWithForm(
+	"#js-edit-profile-picture-modal",
+	handleEditProfilePicture
+);
+
+/****** API CALLS***** */
+
+//CARD SECTION
+
+const opts = {
+	baseUrl: "https://around-api.en.tripleten-services.com/v1",
+	headers: {
+		authorization: "d50bb54b-1efc-4b8a-a5b1-3d5c72d6a1b0",
+		"Content-Type": "application/json",
+	},
+};
+let sectionCards;
+const api = new Api(opts);
+
+/** 1. GET ALL CARDS ROUTE */
+api.getInitialCards()
+	.then((data) => {
+		//handle the response data
+		//console.log("Data from Api: ", data); //debugging statement
+
+		// Create a new Section instance with the fetched data and render the items
+		sectionCards = new Section(
+			{
+				items: data,
+				renderer: createCard,
+			},
+			utils.htmlIds.cardList
+		);
+		sectionCards.renderItems();
+	})
+	.catch((err) => {
+		console.log(err); //debugging statement
+		console.error(`Error getting cards: ${err} `);
+	});
+
+api.getUserInfo()
+	.then((res) => {
+		//console.log("Inside getUserInfo, Response: ", res); //debugging
+		userInfo.setUserInfo({
+			name: res.name,
+			description: res.about,
+		});
+		userInfo.setProfilePicture({
+			profilePicture: res.avatar,
+		});
+	})
+	.catch((err) => {
+		console.log("ERROR: ", err);
+		console.error(err);
+	});
+
+/**
+ * Param: userData
+ * Description:	Sets the change/edit name/title and the description of the profile
+ * 					UPDATE USER PROFILE INFO ROUTE
+ */
+function handleProfileSubmitForm(userData) {
+	// Call the updateUserInfo method from the Api class
+	profileEditForm.renderLoading(true);
+	api.updateUserInfo(userData.title, userData.description)
+		.then(() => {
+			userInfo.setUserInfo({
+				name: userData.title,
+				description: userData.description,
+			});
+			//userInfo.setProfilePicture( {profilePicture: userData.avatar});
+
+			profileEditForm.close();
+		})
+		.catch((err) => {
+			profileEditForm.renderLoading(false);
```

Comment:
`profileEditForm.renderLoading(false);`


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.